### PR TITLE
test: e2e test fix for iOS

### PR DESCRIPTION
**Pull request by Aditi-Singh16**

### CODE REVIEW

File: e2e/03-intraledger-flow.e2e.spec.ts [hunk 1]
```diff
@@ -107,28 +107,28 @@ describe("Username Payment Flow", () => {
...
-    const contexts = await browser.getContexts()
-    const nativeContext = contexts.find((context) =>
-      context.toString().toLowerCase().includes("native"),
-    )
-    await browser.pause(3000)
-    if (nativeContext) {
-      await browser.switchContext(nativeContext.toString())
-    }
+  if (process.env.E2E_DEVICE === "android") {
+    it("Clicks on not enjoying app", async () => {
```
Comment (nicolasburtey) [inline e2e/03-intraledger-flow.e2e.spec.ts:111]:
```suggestion
    it("Dismiss enjoying app popup", async () => {
```

File: e2e/03-intraledger-flow.e2e.spec.ts [hunk 2]
```diff
@@ -107,28 +107,28 @@ describe("Username Payment Flow", () => {
...
-    await browser.pause(3000)
-    const contexts = await browser.getContexts()
-    const nativeContext = contexts.find((context) =>
-      context.toString().toLowerCase().includes("native"),
-    )
-    await browser.pause(3000)
-    if (nativeContext) {
-      await browser.switchContext(nativeContext.toString())
-    }
+  if (process.env.E2E_DEVICE === "android") {
```
Comment (nicolasburtey) [inline e2e/03-intraledger-flow.e2e.spec.ts:110]: why is that needed?

Comment (nicolasburtey) [reply e2e/03-intraledger-flow.e2e.spec.ts:110]: fyi, tests should typically not be skipped like that. it's not a good practice. at a minimum, we should document why we are skipping it if that's necessary

Comment (nicolasburtey) [reply e2e/03-intraledger-flow.e2e.spec.ts:110]: also, why may it pass on other tests, like here? https://github.com/GaloyMoney/galoy-mobile/pull/2499 is it flacky or always not passing?

Comment (Aditi-Singh16) [reply e2e/03-intraledger-flow.e2e.spec.ts:110]: because for ios we have autodismiss alert, so not required for ios
first I thought this was causing the issue, but it was the useEffect

**Comment by sandipndev** [comment]
@Aditi-Singh16 it can happen - I had to retry the e2e test for my branch as well. Let me analyse why that is failing.

File: e2e/03-intraledger-flow.e2e.spec.ts (hunk 2 above)
Comment (nicolasburtey) [reply e2e/03-intraledger-flow.e2e.spec.ts]: > because for ios we have autodismiss alert, so not required for ios

no anymore (afaik)

Comment (Aditi-Singh16) [reply e2e/03-intraledger-flow.e2e.spec.ts]: Yes, I have added it in the feedback flow pr
